### ISSUE TITLE
Incorrect warning on template haskell code

### ISSUE THREAD

**Issue by erikd** [ndmitchell/hlint#436]
The [Hedgehog](https://hackage.haskell.org/package/hedgehog) testing library uses template haskell to find all the properties in a given file (much like QuickCheck) but uses typed quoting:
```
tests :: IO Bool
tests =
  H.checkParallel $$(discover)
```
which HLint incorrectly warns about:
```
test/Test/Jenga/Stack.hs:1:1: Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE TemplateHaskell #-}
Why not remove it.

test/Test/Jenga/Stack.hs:22:21: Warning: Redundant bracket
Found:
  (discover)
Why not:
  discover
```
I suspect this is actually a HSE issue, but raising it here so you can confirm.

Possibly related to #332 .

**Comment by ndmitchell** [comment]
This is a dupe of https://github.com/ndmitchell/hlint/issues/186, although with a better test case, so I've raised that upstream.

However, what this does suggest is that `$(foo)` should be written `$foo` in template haskell world, so I wonder if I should be making that suggestion?

**Comment by erikd** [comment]
If I change the code to:
```
tests :: IO Bool
tests =
  H.checkParallel $$discover
```
as you suggest, I still get:
```
test/Test/Jenga/Stack.hs:1:1: Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE TemplateHaskell #-}
Why not remove it.
```

**Comment by ndmitchell** [comment]
Sorry, I was being confusing. haskell-src-exts can't parse it, and until the linked ticket gets resolved, so until then, you won't have much luck.

Separately, your brackets really are redundant, and I'm wondering if I should warn on `$(foo)` to be `$foo`, and then also give the same warning after HSE supports `$$` too?

**Comment by ndmitchell** [comment]
Confirmed works with HEAD that doesn't use HSE for parsing, so fixed in the next release.
